nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -429,4 +449,53 @@ public void checkTypeParameterNullnessForConditionalExpression(ConditionalExpres
       }
     }
   }
+
+  /**
+   * Checks that for each parameter p at a call, the type parameter nullability for p's type matches
+   * that of the corresponding formal parameter. If a mismatch is found, report an error.
+   *
+   * @param formalParams the formal parameters
+   * @param actualParams the actual parameters
+   * @param isVarArgs true if the call is to a varargs method
+   */
+  public void compareGenericTypeParameterNullabilityForCall(
+      List<Symbol.VarSymbol> formalParams,
+      List<? extends ExpressionTree> actualParams,
+      boolean isVarArgs) {
+    if (!config.isJSpecifyMode()) {
+      return;
+    }
+    for (int i = 0; i < formalParams.size(); i++) {
+      Type formalParameter = formalParams.get(i).type;
+      if (!formalParameter.getTypeArguments().isEmpty()) {
+        Type actualParameter = getTreeType(actualParams.get(i));
+        if (formalParameter instanceof Type.ClassType
+            && actualParameter instanceof Type.ClassType) {
+          if (!compareNullabilityAnnotations(
+              (Type.ClassType) formalParameter, (Type.ClassType) actualParameter)) {
+            reportInvalidParametersNullabilityError(
+                formalParameter, actualParameter, actualParams.get(i), state, analysis);
+          }
+        }
+      }
+    }
+    if (isVarArgs && !formalParams.isEmpty()) {
+      Type.ArrayType varargsArrayType =
+          (Type.ArrayType) formalParams.get(formalParams.size() - 1).type;
+      Type varargsElementType = varargsArrayType.elemtype;
+      if (varargsElementType.getTypeArguments().size() > 0) {
+        for (int i = formalParams.size() - 1; i < actualParams.size(); i++) {
```

Comment:
Doesn't this loop, together with the previous one, double-check the actual parameter at position `formalParams.size() - 1`? Wouldn't this also work with `int i = formalParams.size()` as the initial bound? (Alternatively, the first loop could skip the varargs formal if present)

Reply:
Yes. I will update the first loop. Thanks for the suggestion.